### ISSUE TITLE
Fully autosized DX coil should not fail with a fatal SHR error

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#4853]
User file failing with this error:

   *\* Severe  *\* Coil:Cooling:DX:SingleSpeed "CORRIDOR_FLR_3_DXCOIL" -- coil bypass factor calculation did not converge after max iterations.
   *\*   ~~~   *\* The RatedSHR of [0.576], entered by the user or autosized (see _.eio file),
   *_   ~~~   *\* may be causing this. The line defined by the coil rated inlet air conditions
   *\*   ~~~   *\* (26.7C drybulb and 19.4C wetbulb) and the RatedSHR (i.e., slope of the line) must intersect
   *\*   ~~~   *\* the saturation curve of the psychrometric chart. If the RatedSHR is too low, then this
   *\*   ~~~   *\* intersection may not occur and the coil bypass factor calculation will not converge.
   *\*   ~~~   *\* If autosizing the SHR, recheck the design supply air humidity ratio and design supply air
   *\*   ~~~   *\* temperature values in the Sizing:System and Sizing:Zone objects. In general, the temperatures
   *\*   ~~~   *\* and humidity ratios specified in these two objects should be the same for each system
   *\*   ~~~   *\* and the zones that it serves.
   *\*   ~~~   *\*  During Warmup, Environment=MEMPHIS_TN_USA HEATING 99.6% CONDITIONS, at Simulation time=01/21 00:00 - 00:10
   *\*  Fatal  *\* Coil:Cooling:DX:SingleSpeed "CORRIDOR_FLR_3_DXCOIL" Errors found in calculating coil bypass factors

Two issues here:
1.  If the coil is fully autosized, the autosized SHR should be consistent with the rest of the sizing and this error should not trip.
2.  If the SHR (autosized or hard value) does trip this error, it shouldn't be fatal.  The SHR should be adjusted to something that works and a warning issued, simulation proceeds.

This error was reproduced with Coil:Cooling:DX:SingleSpeed and with Coil:Cooling:DX:TwoSpeed.  Not sure if the code is common or not to all DX coil types.

**Comment by Myoldmopar** [comment]
Please re-test, @mjwitte

**Comment by mjwitte** [comment]
Added another defect file with a hard value for SHR and it still fails but with suspicious outlet temperature.

```
   ** Severe  ** For object = Coil:Cooling:DX:SingleSpeed, name = "CORRIDOR_FLR_3_DXCOIL"
   **   ~~~   ** Calculated outlet air relative humidity greater than 1. The combination of
   **   ~~~   ** rated air volume flow rate, total cooling capacity and sensible heat ratio yields coil exiting
   **   ~~~   ** air conditions above the saturation curve. Possible fixes are to reduce the rated total cooling
   **   ~~~   ** capacity, increase the rated air volume flow rate, or reduce the rated sensible heat ratio for this coil.
   **   ~~~   ** If autosizing, it is recommended that all three of these values be autosized.
   **   ~~~   ** ...Inputs used for calculating cooling coil bypass factor.
   **   ~~~   ** ...Inlet Air Temperature     = 26.67 C
   **   ~~~   ** ...Outlet Air Temperature    = 1.98 C
   **   ~~~   ** ...Inlet Air Humidity Ratio  = 1.125000E-002 kgWater/kgDryAir
   **   ~~~   ** ...Outlet Air Humidity Ratio = 7.027265E-003 kgWater/kgDryAir
   **   ~~~   ** ...Total Cooling Capacity used in calculation = 19503.73 W
   **   ~~~   ** ...Air Mass Flow Rate used in calculation     = 0.543272 kg/s
   **   ~~~   ** ...Air Volume Flow Rate used in calculation   = 0.474578 m3/s
   **   ~~~   **  During Warmup, Environment=MEMPHIS_TN_USA HEATING 99.6% CONDITIONS, at Simulation time=01/21 00:00 - 00:10
   **  Fatal  ** Check and revise the input data for this coil before rerunning the simulation.
```

**Comment by mjwitte** [comment]
And modified and tested the original defect files.  Problem is still there.

**Comment by rraustad** [comment]
PR #5358 

SHR is autosized using an empirical model. This works most times but can lead to an SHR that does not cross the saturation curve and cause CalcCBF to fail.

The fix for this runs CalcCBF silently to iterate on a valid SHR. Usually, the SHR is increased until the ADP temperatures (calculated individually based on T and W using RatedCBF) but there can be times where RH > 1 and SHR is decreased.

```
} else if ( SizingType == CoolingSHRSizing ) {
    if ( DataFlowUsedForSizing >= SmallAirVolFlow && DataCapacityUsedForSizing > 0.0 ) {
        // For autosizing the rated SHR, we set a minimum SHR of 0.676 and a maximum of 0.798. The min SHR occurs occurs at the
        // minimum flow / capacity ratio = MinRatedVolFlowPerRatedTotCap = 0.00004027 [m3/s / W] = 300 [cfm/ton].
        // The max SHR occurs at maximum flow / capacity ratio = MaxRatedVolFlowPerRatedTotCap = 0.00006041 [m3/s / W] = 450 [cfm/ton].
        // For flow / capacity ratios between the min and max we linearly interpolate between min and max SHR. Thus rated SHR is a
        // linear function of the rated flow / capacity ratio. This linear function (see below) is the result of a regression
        // of flow/capacity ratio vs SHR for several actual coils.
        RatedVolFlowPerRatedTotCap = DataFlowUsedForSizing / DataCapacityUsedForSizing;
        if ( DXCT == RegularDXCoil ) {
            if ( RatedVolFlowPerRatedTotCap > MaxRatedVolFlowPerRatedTotCap( DXCT ) ) {
                AutosizeDes = 0.431 + 6086.0 * MaxRatedVolFlowPerRatedTotCap ( DXCT );
            } else if ( RatedVolFlowPerRatedTotCap < MinRatedVolFlowPerRatedTotCap( DXCT ) ) {
                AutosizeDes = 0.431 + 6086.0*MinRatedVolFlowPerRatedTotCap ( DXCT );
            } else {
                AutosizeDes = 0.431 + 6086.0*RatedVolFlowPerRatedTotCap;
            }
        } else { // DOASDXCoil, or DXCT = 2
            if ( RatedVolFlowPerRatedTotCap > MaxRatedVolFlowPerRatedTotCap( DXCT ) ) {
                AutosizeDes = 0.389 + 7684.0*MaxRatedVolFlowPerRatedTotCap ( DXCT );
```

**Comment by Myoldmopar** [comment]
Closed via #5358
